sample/hardhat.config.ts

```diff
@@ -1,76 +1,49 @@
-import { HardhatUserConfig } from "hardhat/config";
-import "@nomicfoundation/hardhat-toolbox";
-import "hardhat-multibaas-plugin";
-import "@openzeppelin/hardhat-upgrades";
-import path from "path";
+import { defineConfig } from "hardhat/config";
+import hardhatMultiBaasPlugin from "hardhat-multibaas-plugin";
+import path from "node:path";
+import { fileURLToPath, pathToFileURL } from "node:url";
 
-let deployerPrivateKey =
-  "0x0000000000000000000000000000000000000000000000000000000000000000";
-let deploymentEndpoint, ethChainID, web3Key, adminApiKey;
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
 
-if (process.env.HARDHAT_NETWORK) {
-  const CONFIG_FILE = path.join(
-    __dirname,
-    `./deployment-config.${process.env.HARDHAT_NETWORK}`
-  );
-  ({
-    deploymentConfig: {
-      deploymentEndpoint,
-      ethChainID,
-      deployerPrivateKey,
-      web3Key,
-      adminApiKey,
-    },
-  } = require(CONFIG_FILE));
-}
+const networkName = process.env.HARDHAT_NETWORK ?? "development";
+const configPath = path.resolve(
+  __dirname,
+  `./deployment-config.${networkName}.ts`,
+);
+
+const { deploymentConfig } = (await import(pathToFileURL(configPath).href)) as {
+  deploymentConfig: {
+    deploymentEndpoint: string;
+    ethChainID: number;
+    web3Key: string;
+    adminApiKey: string;
+    deployerPrivateKey: string;
+  };
+};
 
-const config: HardhatUserConfig = {
+export default defineConfig({
+  plugins: [hardhatMultiBaasPlugin],
+  solidity: {
+    version: "0.8.28",
+    npmFilesToBuild: [
+      "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol",
+      "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol",
+    ],
+  },
   networks: {
     development: {
-      url: `${deploymentEndpoint}/web3/${web3Key}`,
-      chainId: ethChainID,
-      accounts: [deployerPrivateKey],
+      type: "http",
+      chainType: "l1",
+      url: `${deploymentConfig.deploymentEndpoint}/web3/${deploymentConfig.web3Key}`,
+      chainId: deploymentConfig.ethChainID,
+      accounts: [deploymentConfig.deployerPrivateKey],
```

Comment:
Adding this skips waiting for 5 blocks, where on non-HRE local dev networks with insta-mining, requires mining 5 more blocks manually.

Note that this change as-is works functionally, but will result in a linting error which will need to be resolved.

```suggestion
      accounts: [deploymentConfig.deployerPrivateKey],
      ignition: {
        requireConfirmations: 1,
      },
```

Reply:
https://github.com/curvegrid/hardhat-multibaas-plugin/pull/807/commits/6439fa177b1964f0e0907592ddbb38f04721e89c

https://hardhat.org/ignition/docs/reference/config#requiredconfirmations shows `{ ignition: requiredConfirmations }` looks at the top-level

Reply:
Yes, but the point is that we want to wait for a single confirmation only on local development. On other chains, it might be advisable to wait for the default number of blocks (5?) to avoid issues with eventual consistency. WDYT?

Reply:
I have double checked `requiredConfirmations` option, but it did not work when I added it in to `development` but worked when I added it at top-level. The `ignition` option in `development` is ignored. 🤔 

```ts
  // === CASE 1: WORK
  // ignition: {
  //   requiredConfirmations: 1,
  // },
  networks: {
    development: {
      type: "http",
      chainType: "l1",
      url: `${deploymentConfig.deploymentEndpoint}/web3/${deploymentConfig.web3Key}`,
      chainId: deploymentConfig.ethChainID,
      accounts: [deploymentConfig.deployerPrivateKey],
      // === CASE 2: NOT WORK
      // ignition: {
      //   requiredConfirmations: 1,
      // },
    },
  },
```

### CASE 1: WORK

See the end of log

```sh
START=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
npm run deploy:greeter -- --reset
END=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
echo "start=${START} end=${END}"

> hardhat-multibaas-plugin-sample-v3@1.0.0 deploy:greeter
> npm run deploy:base -- ignition/modules/GreeterModule.ts --reset


> hardhat-multibaas-plugin-sample-v3@1.0.0 deploy:base
> HARDHAT_NETWORK=development HARDHAT_IGNITION_CONFIRM_DEPLOYMENT=false npx hardhat ignition deploy ignition/modules/GreeterModule.ts --reset

✔ Confirm reset of deployment "chain-1337" on chain 1337? … yes
Hardhat Ignition 🚀

Deploying [ GreeterModule ]

Batch #1
  Executed GreeterModule#Greeter

[ GreeterModule ] successfully deployed 🚀

Deployed Addresses

GreeterModule#Greeter - 0x79fdB736e52b591ef6272C04A0B6E8F690d5786F
MultiBaas: Contract "Greeter 1.0" already created. Skipping creation.
MultiBaas: Deleting old address 0x070C374dfd107986F95d53264659a2e2B119eCe3 with alias "greeter"
MultiBaas: Creating address 0x79fdB736e52b591ef6272C04A0B6E8F690d5786F with alias "greeter"
MultiBaas: Linking contract "greeter 1.0" to address "greeter"
start=2025-12-30T10:32:42Z end=2025-12-30T10:32:53Z ### ABOUT 10 seconds (1 block)
```

### CASE 2: NOT WORK

See the end of log

```sh
START=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
npm run deploy:greeter -- --reset
END=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
echo "start=${START} end=${END}"

> hardhat-multibaas-plugin-sample-v3@1.0.0 deploy:greeter
> npm run deploy:base -- ignition/modules/GreeterModule.ts --reset


> hardhat-multibaas-plugin-sample-v3@1.0.0 deploy:base
> HARDHAT_NETWORK=development HARDHAT_IGNITION_CONFIRM_DEPLOYMENT=false npx hardhat ignition deploy ignition/modules/GreeterModule.ts --reset

✔ Confirm reset of deployment "chain-1337" on chain 1337? … yes
Hardhat Ignition 🚀

Deploying [ GreeterModule ]

Batch #1
  Executed GreeterModule#Greeter

[ GreeterModule ] successfully deployed 🚀

Deployed Addresses

GreeterModule#Greeter - 0x070C374dfd107986F95d53264659a2e2B119eCe3
MultiBaas: Contract "Greeter 1.0" already created. Skipping creation.
MultiBaas: Deleting old address 0xc6952ca5C0C4Dab0EC732959958356F55c8Ca358 with alias "greeter"
MultiBaas: Creating address 0x070C374dfd107986F95d53264659a2e2B119eCe3 with alias "greeter"
MultiBaas: Linking contract "greeter 1.0" to address "greeter"
start=2025-12-30T10:30:26Z end=2025-12-30T10:31:12Z ### ABOUT 50 seconds (5 blocks)
```

### SUGGESTION

What do you think of this?

```ts
  ignition: {
    requiredConfirmations: networkName === "development" ? 1  : 5,
  },
  networks: {
    development: {
      type: "http",
      chainType: "l1",
      url: `${deploymentConfig.deploymentEndpoint}/web3/${deploymentConfig.web3Key}`,
      chainId: deploymentConfig.ethChainID,
      accounts: [deploymentConfig.deployerPrivateKey],
    },
  },
```

Reply:
Crazy...I could of sworn I had this working locally, but I reverted to pull in your changes to review. Now I can't get it working on local again. Sorry for sending you down a rabbit hole on this...let's just go with:

```
  ignition: {
    requiredConfirmations: 1, // CAUTION: suitable for development environments only with on-demand block production
  },
```

Reply:
Thank you for clarifying! Happy to go with it!